hydromt/io.py

```diff
@@ -615,13 +616,12 @@ def open_vector(
     gdf : geopandas.GeoDataFrame
         Parsed geometry file
     """
-    filtered = False
     driver = driver if driver is not None else str(fn).split(".")[-1].lower()
     if driver in ["csv", "parquet", "xls", "xlsx", "xy"]:
         gdf = open_vector_from_table(fn, driver=driver, **kwargs)
     else:
-        gdf = gpd.read_file(fn, bbox=bbox, mask=geom, mode=mode, **kwargs)
-        filtered = predicate == "intersects"
+        bbox_reader = gis_utils.prepare_pyogrio_reader_filters(fn, bbox, geom, crs)
+        gdf = read_dataframe(fn, bbox=bbox_reader, mode=mode, **kwargs)
```

Comment:
Instead of `pyogrio.read_dataframe` have you also tried `gpd.read_file(..., engine='pyogrio')`? I haven't tried this, but that option would potentially solve your bbox argument issue? This could also be configured from the DataCatalog `driver_kwargs`

```yaml
data_type: GeoDataFrame
driver: vector
driver_kwargs:
   engine: pyogrio
```
Also, before we replace `gpd.read_file` (with fiona as default reader) it would be good check if all vector file formats commonly used in hydromt are supported by pyogrio. Without haven't looked into this I would still be in favor of keeping both flavour next to each other for now. The `engine` argument in gpd.read_file seems to enable that.

Reply:
The issue that this also fixes the fact that geopandas calls `Path.expanduser`, which for UPath results in a `NotImplementedError`, while pyogrio does not. As an alternative I can also subclass UPath and implement os.expanduser?

Reply:
The `mask` argument is not yet implemented in geopandas

Reply:
I think `gpd.read_file` does except file-like objects. Perhaps opening the UPath as a file-like object (see lines below) could also work to solve the `Path.expanduser` issue? Converting the `UPath` object to a `str` looses the `storage_options` arguments which I think we may need down the road to properly access cloud data.

```
with fsspec.open(fn) as fobj:
    gpd.read_file(fobj, engine='pyogrio')
````